crates/router/src/core/payments/operations/payment_get_intent.rs

```diff
@@ -0,0 +1,247 @@
+        let payment_intent = db
+            .find_payment_intent_by_id(key_manager_state, &request.id, key_store, storage_scheme)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Payment Intent Not Found")?;
+
+        let payment_data = payments::PaymentIntentData {
+            flow: PhantomData,
+            payment_intent,
+        };
+
+        let get_trackers_response = operations::GetTrackerResponse {
+            operation: Box::new(self),
+            payment_data,
+        };
+
+        Ok(get_trackers_response)
+    }
+}
+
+#[async_trait]
+impl<F: Clone> UpdateTracker<F, payments::PaymentIntentData<F>, PaymentsGetIntentRequest>
+    for PaymentGetIntent
+{
+    #[instrument(skip_all)]
+    async fn update_trackers<'b>(
+        &'b self,
+        _state: &'b SessionState,
+        _req_state: ReqState,
+        payment_data: payments::PaymentIntentData<F>,
+        _customer: Option<domain::Customer>,
+        _storage_scheme: enums::MerchantStorageScheme,
+        _updated_customer: Option<storage::CustomerUpdate>,
+        _key_store: &domain::MerchantKeyStore,
+        _frm_suggestion: Option<FrmSuggestion>,
+        _header_payload: api::HeaderPayload,
+    ) -> RouterResult<(
+        PaymentsGetIntentOperation<'b, F>,
+        payments::PaymentIntentData<F>,
+    )>
+    where
+        F: 'b + Send,
+    {
+        Ok((Box::new(self), payment_data))
+    }
+}
+
+impl<F: Send + Clone> ValidateRequest<F, PaymentsGetIntentRequest, payments::PaymentIntentData<F>>
+    for PaymentGetIntent
+{
+    #[instrument(skip_all)]
+    fn validate_request<'a, 'b>(
+        &'b self,
+        _request: &PaymentsGetIntentRequest,
+        merchant_account: &'a domain::MerchantAccount,
+    ) -> RouterResult<(
+        PaymentsGetIntentOperation<'b, F>,
+        operations::ValidateResult,
+    )> {
+        Ok((
+            Box::new(self),
+            operations::ValidateResult {
+                merchant_id: merchant_account.get_id().to_owned(),
+                storage_scheme: merchant_account.storage_scheme,
+                requeue: false,
+            },
+        ))
+    }
+}
+
+#[async_trait]
+impl<F: Clone + Send> Domain<F, PaymentsGetIntentRequest, payments::PaymentIntentData<F>>
+    for PaymentGetIntent
+{
+    #[instrument(skip_all)]
+    async fn get_customer_details<'a>(
+        &'a self,
+        state: &SessionState,
+        payment_data: &mut payments::PaymentIntentData<F>,
+        merchant_key_store: &domain::MerchantKeyStore,
+        storage_scheme: enums::MerchantStorageScheme,
+    ) -> CustomResult<
+        (
+            BoxedOperation<'a, F, PaymentsGetIntentRequest, payments::PaymentIntentData<F>>,
+            Option<domain::Customer>,
+        ),
+        errors::StorageError,
+    > {
+        // validate customer_id if sent in the request
+        if let Some(id) = payment_data.payment_intent.customer_id.clone() {
```

Comment:
This shouldn't be needed for `GetIntent`. Should I remove this?

Reply:
You can remove. if any use case come we can add back